From eb2bedfbef86f4144f069a244a9c495704c5dfe8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Marc-Andr=C3=A9=20Lureau?= <marcandre.lureau@redhat.com>
Date: Tue, 29 Oct 2013 13:02:07 +0200
Subject: [PATCH] window: move window again after fullscreen

Some WM get fullscreen position wrong, in some racy ways (metacity on
rhel6 for instance).  Setting window position after entering fullscreen
seems to solve the problem. Though it's better to move before entering
fullscreen, to avoid extra resizes.

https://bugzilla.redhat.com/show_bug.cgi?id=809546
---
 src/virt-viewer-window.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/virt-viewer-window.c b/src/virt-viewer-window.c
index 5d646ac..9ab75a1 100644
--- a/src/virt-viewer-window.c
+++ b/src/virt-viewer-window.c
@@ -538,6 +538,9 @@ virt_viewer_window_enter_fullscreen(VirtViewerWindow *self, gint monitor)
     virt_viewer_window_move_to_monitor(self);
 
     gtk_window_fullscreen(GTK_WINDOW(priv->window));
+
+    /* move again, some WM get fullscreen position wrong, in some racy ways.. */
+    virt_viewer_window_move_to_monitor(self);
 }
 
 #define MAX_KEY_COMBO 3
